hydra-node/exe/hydra-node/Main.hs

```diff
@@ -110,6 +126,29 @@ main = do
 
     action (Ledger.cardanoLedger globals ledgerEnv)
 
+  -- check if hydra-node parameters are matching with the hydra-node state.
+  -- REVIEW: Should we also check against items we receive on-chain here? e.g. 'OpenThreadOutput' if in Open state?
+  checkParamsAgainstExistingState :: HeadState Ledger.Tx -> Environment -> [String]
+  checkParamsAgainstExistingState hs env =
+    case hs of
+      Idle _ -> []
+      Initial InitialState{parameters} -> validateParameters "InitialState: " parameters
+      Open OpenState{parameters} -> validateParameters "OpenState: " parameters
+      Closed ClosedState{parameters} -> validateParameters "ClosedState: " parameters
+   where
+    validateParameters st params =
+      let res = flip execState [] $ do
```

Comment:
why do we need State here?

would not be better to use Writer instead, so we can accumulate all the miss configs?